lib/puppet-lint/plugins/check_deprecated_functions/deprecated_functions.rb

```diff
@@ -0,0 +1,27 @@
+# Public: A puppet-lint custom check to detect deprecated functions.
+DEPRECATED_FUNCTIONS_VAR_TYPES = Set[:FUNCTION_NAME]
```

Comment:
These functions are also available as built in ones. How do you distinguish them? In fact, latest stdlib has removed some of those deprecated ones that are still available. The `FUNCTION_NAME` in puppet-lint is just a simple regular expression, matching both built in and stdlib defined ones.

Reply:
This was originally conceived as a one of check while updating, to draw attention to any possible violations.
Considering this it may be best to move it out of the main puppet-lint and into its own separate gem that can be included as needed and then discarded.
I don't foresee this as a permanent check due to the issue you mentioned above, that functions are also available as build in ones and the lack of ability within puppet-lint to tell them apart.